Assets/HoloToolkit/Input/Scripts/KeywordManager.cs

```diff
@@ -84,26 +102,30 @@ void OnDestroy()
             }
         }
 
+        /// <summary>
+        /// Listens for key presses and calls the corrisponding keyword event
+        /// </summary>
         private void ProcessKeyBindings()
         {
-            foreach (var kvp in KeywordsAndResponses)
+            for (int i = 0; i < KeywordsAndResponses.Length; i++)
```

Comment:
I agree! This implementation has been on my list of things to look at since it was added to the toolkit. It would probably be worth wrapping the call in Update with
`if (Input.anyKeyDown)`
so it's only called [during the frame a key is actually pressed](https://docs.unity3d.com/ScriptReference/Input-anyKeyDown.html), that way this loop will only happen when necessary instead of every frame.